packages/react/src/components/LocaleModal/LocaleModal.js

```diff
@@ -0,0 +1,165 @@
+    (async () => {
+      const locale = await LocaleAPI.getLocale();
+      const list = locale && (await LocaleAPI.getList(locale));
+      setList(list);
+    })();
+  }, []);
+
+  /**
+   *  New region/country list based lang attributes available on page
+   *
+   * @param {object} list country list
+   *
+   * @returns {object} list item
+   */
+  const sortList = list => {
+    const pageLangs = altlangs();
+    const filterList = [];
+
+    list.regionList &&
+      list.regionList.map((region, index) => {
+        filterList.push({
+          name: region.name,
+          key: region.key,
+          countries: [],
+        });
+
+        for (let [key, value] of Object.entries(pageLangs)) {
+          region.countryList.map(country => {
+            country.locale.map(loc => {
+              if (loc[0].includes(key)) {
+                filterList[index].countries.push({
+                  region: region.key,
+                  name: country.name,
+                  locale: loc[0],
+                  language: loc[1],
+                  href: value,
+                });
+              }
+            });
+          });
+        }
+
+        filterList[index].countries.sort((a, b) => (a.name > b.name ? 1 : -1));
+      });
+
+    return filterList;
+  };
+
+  return (
+    <ComposedModal
+      open={isOpen}
+      onClose={close}
+      data-autoid={`${stablePrefix}--locale-modal`}>
+      {isFiltering ? (
+        <ModalHeader
+          data-autoid={`${stablePrefix}--locale-modal__region-back`}
+          label={[
+            <ArrowLeft20 className={`${prefix}--locale-modal__label-arrow`} />,
+            localeModalProps.headerTitle,
+          ]}
+          title={currentRegion}
+          className={`${prefix}--locale-modal__back`}></ModalHeader>
+      ) : (
+        <ModalHeader
+          label={[
+            localeModalProps.headerLabel,
+            <Globe20 className={`${prefix}--locale-modal__label-globe`} />,
+          ]}
+          title={localeModalProps.headerTitle}
+        />
+      )}
+      <ModalBody className={`${prefix}--locale-modal ${filterClass}`}>
+        <LocaleModalRegions
+          regionList={sortList(list)}
+          setCurrentRegion={setCurrentRegion}
+          setIsFiltering={setIsFiltering}
+          {...localeModalProps}
+        />
+        <LocaleModalCountries
+          regionList={sortList(list)}
+          setIsFiltering={setIsFiltering}
+          {...localeModalProps}
+        />
+      </ModalBody>
+    </ComposedModal>
+  );
+
+  /**
+   * Sets modal state to closed
+   *
+   * @private
+   */
+  function close() {
+    setIsOpen(false);
+  }
+};
+
+/**
+ * @property propTypes
+ * @description Defined property types for component
+ * @type {{isOpen: boolean, setIsOpen: boolean, headerLabel: string, headerTitle: string}}
+ */
+LocaleModal.propTypes = {
+  isOpen: PropTypes.bool,
+  setIsOpen: PropTypes.bool,
+  headerLabel: PropTypes.string,
+  headerTitle: PropTypes.string,
```

Comment:
@kennylam The `headerTitle` will need to be set based on the `getLang` display name, as well as in the button label in the footer. If this can't make code freeze, we at least need to track this as a hotfix before we go live with `v1.2.0`.

cc: @annawen1

Reply:
@jeffchew yes all of these are going to be put into the translation files and were not captured in the scope of this story. Need to get all of translatable fields into the json